src/components/pages/OnboardPage.tsx

```diff
@@ -1,38 +1,34 @@
-import React, { useState } from "react";
+import React, { useEffect, useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
 import Hero_Image from "@app/assets/onboard_hero_image.jpg";
 import Meshtastic_Logo from "@app/assets/Mesh_Logo_Black.png";
 import SerialPortOption from "../Onboard/SerialPortOption";
 
-type PortState = "IDLE" | "LOADING" | "SUCCESS" | "FAILURE";
+import { requestAvailablePorts } from "@features/device/deviceActions";
+import { selectAvailablePorts } from "@app/features/device/deviceSelectors";
+
+type PortState = "IDLE" | "PENDING" | "SUCCESS" | "FAILED";
 type PortType = { name: string; state: PortState };
 
 export interface IOnboardPageProps {
   unmountSelf: () => void;
 }
 
 const OnboardPage = ({ unmountSelf }: IOnboardPageProps) => {
-  const mockPorts: PortType[] = [
-    {
-      name: "port1",
-      state: "FAILURE",
-    },
-    {
-      name: "port2",
-      state: "IDLE",
-    },
-    {
-      name: "port3",
-      state: "SUCCESS",
-    },
-    {
-      name: "port4",
-      state: "LOADING",
-    },
-  ];
+  const availableSerialPorts = useSelector(selectAvailablePorts());
+  const dispatch = useDispatch();
+  const [activeNode, setActiveNode] = useState(-1);
```

Comment:
Might just use `null` for this, makes life a bit easier in TS

Reply:
Beginner question -- what's standard for hooks? empty string or `null`?

Reply:
Updated.